drivers/azure/azureutil/securityrules.go

```diff
@@ -0,0 +1,313 @@
+			openPortsToBeDeleted[to.String(rule.Name)] = rule
+		}
+	}
+	// Remove from deletion if it exists or add to queue if it does not exist
+	openPortsToBeAdded := make(map[string]network.SecurityRule)
+	for _, p := range openPorts {
+		op := parseOpenPort(p)
+		if op == nil {
+			return fmt.Errorf("unable to parse open port %s", p)
+		}
+		if _, ok := openPortsToBeDeleted[op.name]; ok {
+			// Do not delete the open port
+			delete(openPortsToBeDeleted, op.name)
+			continue
+		}
+		openPortsToBeAdded[op.name] = network.SecurityRule{
+			Name: to.StringPtr(op.name),
+			SecurityRulePropertiesFormat: &network.SecurityRulePropertiesFormat{
+				Description:              to.StringPtr("User requested port to be accessible from Internet via docker-machine"),
+				SourceAddressPrefix:      to.StringPtr("*"),
+				DestinationAddressPrefix: to.StringPtr("*"),
+				SourcePortRange:          to.StringPtr("*"),
+				DestinationPortRange:     to.StringPtr(op.port),
+				Access:                   network.SecurityRuleAccessAllow,
+				Direction:                network.SecurityRuleDirectionInbound,
+				Protocol:                 op.proto,
+			},
+		}
+	}
+	// Log open ports to be deleted and added
+	if len(openPortsToBeDeleted) > 0 || len(openPortsToBeAdded) > 0 {
+		removedPorts := make([]string, len(openPortsToBeDeleted))
+		addedPorts := make([]string, len(openPortsToBeAdded))
+		i := 0
+		for op := range openPortsToBeDeleted {
+			removedPorts[i] = op
+			i++
+		}
+		i = 0
+		for op := range openPortsToBeAdded {
+			addedPorts[i] = op
+			i++
+		}
+		log.Info("Updating security rule configuration", logutil.Fields{
+			"openPortsRemoved": removedPorts,
+			"openPortsAdded":   addedPorts,
+		})
+	}
+	// Remove all unnecessary open ports and release the priorities
+	for _, rule := range openPortsToBeDeleted {
+		delete(takenPriorities, to.Int32(rule.Priority))
+		removeSecurityRule(rules, rule)
+	}
+	// Add any necessary open ports
+	priority := int32(1000)
+	for _, rule := range openPortsToBeAdded {
+		// Find a priority that can be given to this rule
+		_, taken := takenPriorities[priority]
+		for taken {
+			priority++
+			_, taken = takenPriorities[priority]
+		}
+		rule.Priority = to.Int32Ptr(priority)
+		priority++
+		*rules = append(*rules, rule)
+	}
+	return nil
+}
+
+// checkIfDockerConflict does a rudimentary check to see if the rule could potentially conflict with the ability
+// for the docker port to be accessible. If the rule does not pass all rudimentary checks, it returns an error.
+func checkIfDockerConflict(rule network.SecurityRule, dockerPort int) error {
+	f := logutil.Fields{
+		"name":        to.String(rule.Name),
+		"description": to.String(rule.Description),
+		"priority":    to.Int32(rule.Priority),
+		"access":      rule.Access,
+		"direction":   rule.Direction,
+		"protocol":    rule.Protocol,
+		"portRanges":  getDestinationPorts(rule),
+	}
+	priority := to.Int32(rule.Priority)
+	if priority > dockerPriority {
```

Comment:
we only call `checkIfDockerConflict` if priority < dockerPriority, so do we need this check? if we still do can we move before line 206 and return if needed

Reply:
Good catch, we can omit this check here